### PR TITLE
chore(mantine): button and action icon overridable via theme

### PR DESCRIPTION
**Pull request by GermainBergeron**
### Proposed Changes

Made ActionIcon and Button theme-able via the factory
Extracted duplicated code between ActionIcon and Button to a shared hook

### Potential Breaking Changes

None?

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Review by toofff — APPROVED** [review]

File: packages/mantine/src/hooks/useClickWithLoading.ts
Comment (toofff) [inline packages/mantine/src/hooks/useClickWithLoading.ts:1]: Maybe add some documentation if you want to export these hooks outside the lib?

**Review by FelixBlaisThon — APPROVED** [review]
I see that we dont spread getStyle in the component, I guess it's because we are just wrapping the actual mantine button which will handle all of it behind the couvarte ?

**Comment by GermainBergeron** [comment]
> I see that we dont spread getStyle in the component, I guess it's because we are just wrapping the actual mantine button which will handle all of it behind the couvarte ?

Exactly, we're passing all props to the original component

File: packages/mantine/src/hooks/useClickWithLoading.ts
Comment (GermainBergeron) [reply packages/mantine/src/hooks/useClickWithLoading.ts:1]: I haven't exported the hook, for now we'll keep it internal

**Pull request merged**